crescent/plugin.py

```diff
@@ -44,12 +48,27 @@ def load(self, path: str, refresh: bool = False) -> Plugin:
 
         Args:
             path: The module path for the plugin.
-            refresh: Whether or not to reload the plugin.
+            refresh: Whether or not to reload the plugin and the plugin's module.
         """
 
-        plugin = Plugin._from_module(path, refresh=refresh)
-        self.add_plugin(plugin, force=refresh)
-        return plugin
+        new_plugin = Plugin._from_module(path, refresh=refresh)
+
+        if refresh:
+            old_plugin = self.plugins.pop(new_plugin.name)
+            old_plugin._unload()
+
+        self._add_plugin(new_plugin, refresh=refresh)
+
+        return new_plugin
```

Comment:
Realizing that this logic will fail if the plugin name changed, since `Plugin._from_module` will return the new plugin name (since you may have passed refresh=True).

```py
if refresh:
    self.plugins.pop(Plugin._from_module(path, refresh=False))._unload()
    new_plugin = Plugin._from_module(path, refresh=True)
else:
    new_plugin = Plugin._from_module(path, refresh=False)

self._add_plugin(new_plugin, refresh=refresh)
return new_plugin
```

Reply:
Had this realization once I went to bed lol